'make' does not recognize that compilation did not succeed

On my system, after calling `./configure --FC=gfortran --CC=gcc` and then `make`, this is the output:
```
Cleaning up existing executables in src/build/compiled...
 
******************************************************************** 
                 Now Building: /rayleigh.opt
 
                  
 
Cleaning up compiled object files...
gfortran -O3 -ffixed-line-length-132 -I/include -c Structures.F90
f951: Warning: Nonexistent include directory ‘/include’ [-Wmissing-include-dirs]
gfortran -O3 -ffixed-line-length-132 -I/include -c BufferedOutput.F90
f951: Warning: Nonexistent include directory ‘/include’ [-Wmissing-include-dirs]
gcc -w    -c -o cmkdir.o cmkdir.c
gfortran -O3 -ffixed-line-length-132 -I/include -c MakeDir.F90
f951: Warning: Nonexistent include directory ‘/include’ [-Wmissing-include-dirs]
gfortran -O3 -ffixed-line-length-132 -I/include -c MPI_Base.F90
f951: Warning: Nonexistent include directory ‘/include’ [-Wmissing-include-dirs]
MPI_Base.F90:8: Error: Can't open included file 'mpif.h'
make[3]: *** [MPI_Base.o] Error 1
make[2]: *** [opt] Error 2
 
 
******************************************************************** 
                 Now Building: /rayleigh.dbg
 
                  
 
Cleaning up compiled object files...
gfortran -O0 -fbounds-check -fbacktrace -ffixed-line-length-132 -I/include -c Structures.F90
f951: Warning: Nonexistent include directory ‘/include’ [-Wmissing-include-dirs]
gfortran -O0 -fbounds-check -fbacktrace -ffixed-line-length-132 -I/include -c BufferedOutput.F90
f951: Warning: Nonexistent include directory ‘/include’ [-Wmissing-include-dirs]
gcc -w    -c -o cmkdir.o cmkdir.c
gfortran -O0 -fbounds-check -fbacktrace -ffixed-line-length-132 -I/include -c MakeDir.F90
f951: Warning: Nonexistent include directory ‘/include’ [-Wmissing-include-dirs]
gfortran -O0 -fbounds-check -fbacktrace -ffixed-line-length-132 -I/include -c MPI_Base.F90
f951: Warning: Nonexistent include directory ‘/include’ [-Wmissing-include-dirs]
MPI_Base.F90:8: Error: Can't open included file 'mpif.h'
make[3]: *** [MPI_Base.o] Error 1
make[2]: *** [dbg] Error 2
 

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Compilation is complete.
 
 
Run 'make install' to install the rayleigh executables into: 
/home/fac/f/bangerth/p/deal.II/1/projects/Rayleigh/bin
```
Only looking at the last few lines suggests that compilation succeeded, and indeed the return code is zero. But that's not correct, of course: Compilation actually failed. Failure to compile a file should terminate the entire operation.

--- a/src/Makefile
+++ b/src/Makefile
@@ -43,8 +43,7 @@ default:
 
 all: 
 	@for i in $(VERSIONS) ; do \
-		$(MAKE) $$i VERSION=$$i PREFIX=$(PREFIX); \
-		echo ' ' ; \
+		$(MAKE) $$i VERSION=$$i PREFIX=$(PREFIX) && echo ' ' ; \
 	done
 
 rayleigh_exec: $(OBJ)

